src/business-logic/services/test/device-service.spec.ts

```diff
@@ -0,0 +1,41 @@
+import {DeviceServiceImplementation} from '../device-service-implementation'
+import {Device, TelemetricsDevice} from '../../../model/entities/device'
+import {instance, mock, verify, when} from '@typestrong/ts-mockito'
+import {MongoDeviceRepository} from '../../../data-access/repositories/mongo/mongo-device-repository'
+import {DeviceService} from '../interfaces/device-service'
+import {DeviceEventEmitter} from '../interfaces/device-event-emitter'
+
+describe('DeviceService', () => {
+  let testDevice: TelemetricsDevice
+  let deviceRepository: MongoDeviceRepository
+  let deviceEventEmitter: DeviceEventEmitter
+  const mockedValue: MongoDeviceRepository = mock<MongoDeviceRepository>()
+
+  beforeEach(() => {
+    testDevice = mock<TelemetricsDevice>()
+    deviceRepository = instance(mockedValue)
+    deviceEventEmitter = mock<DeviceEventEmitter>()
+  })
+
+
+  it('should read all configurations', async () => {
+    const devices: Device[] = [testDevice]
+    const mockedService: DeviceService = mock<DeviceService>()
+    const testee: DeviceServiceImplementation = new DeviceServiceImplementation(deviceRepository, deviceEventEmitter)
+    when(mockedService.getDevices()).thenReturn(Promise.resolve(devices))
+
+    await testee.getDevices()
+
+    verify(mockedValue.getDevices()).called()
```

Comment:
Line 22, 23 and 25 has no effect on the test and the test still parses without it.
![image](https://github.com/user-attachments/assets/0bf0ccaa-5499-40d5-bcee-bb495f9d94d3)

Also, your test is testing implementation, not functionality. If I simply return an empty array in the method, your test still parse, but the method no longer works.
![image](https://github.com/user-attachments/assets/a622f81d-f088-4cce-b133-71c82e2d1d1d)


When writing test for a method that returns something, we should make our assertions on the returned valued so we can be sure that we are getting the correct result. Our tests shouldn't care about how we got the correct result (i.e. implementation) as long as we get the correct result.
In short, that means we should make our assertions on the returned value. That could look like this:
![image](https://github.com/user-attachments/assets/49a8843d-d523-424a-a79f-0542540b1d98)

Reply:
This standard has been applied. For this PR I added the getDevice, getDevices and create tests, and the rest of the controller tests are submitted with the SOF-1968 PR branch